source/vibe/stream/openssl.d

```diff
@@ -469,6 +471,17 @@ final class OpenSSLContext : SSLContext {
 			SSL_CTX_set_cipher_list(m_ctx, toStringz(list));
 	}
 
+	/** Make up a context ID to assign to the SSL context.
+
+		This is very required when doing client cert authentication,
+		otherwise many connections will go aborted.
+	*/
+	void setContextID()
```

Comment:
Since it is called in the constructor already, does this method have to be public? If yes, I'd change the name to `setSessionIDContext` to match the semantics more closely. What I didn't understand yet when reading the documentation of `SSL_CTX_set_session_id_context` is if using the host name as the context name is the right choice. From the description it sounds like it should rather be an application specific string, which would have to be passed as a parameter.